### ISSUE TITLE
missing_docs false positive on re-export in module with #![allow(missing_docs)]

### ISSUE THREAD

**Issue by abonander** [rust-lang/rust#57533]
This one might be a bit complex to solve. This affects both 2015 and 2018 editions on stable/nightly.

When a `pub` item from a private module in a crate with `#![deny(missing_docs)]` is reexported in a pub module marked `#![allow(missing_docs)]`, the lint still fires.

[2015 (Playground)](https://play.rust-lang.org/?version=stable&mode=debug&edition=2015&gist=06f58f7154d4a917afb2d9a366a245a3) (2018 just needs the `crate::` prefix on the `use` path)
```rust
#![deny(missing_docs)]

#[allow(missing_docs)]
pub mod undoc {
    #[allow(missing_docs)] // no effect
    pub use nonpub::Foo;
}

mod nonpub {
    pub struct Foo;
}
```

Both report the same error. The lint marks the original definition point of the item; if it has methods, those are hit too:
```
error: missing documentation for crate
  --> src/lib.rs:1:1
   |
1  | / #![deny(missing_docs)]
2  | |
3  | | #[allow(missing_docs)]
4  | | pub mod undoc {
...  |
14 | |     }
15 | | }
   | |_^
   |
note: lint level defined here
  --> src/lib.rs:1:9
   |
1  | #![deny(missing_docs)]
   |         ^^^^^^^^^^^^

error: missing documentation for a struct
  --> src/lib.rs:10:5
   |
10 |     pub struct Foo;
   |     ^^^^^^^^^^^^^^^

error: missing documentation for a method
  --> src/lib.rs:13:9
   |
13 |         pub fn bar() {}
   |         ^^^^^^^^^^^^
```

**Comment by abonander** [comment]
@estebank this is a lint error, not a Rustdoc error

**Comment by estebank** [comment]
@abonander good catch, I misread the ticket while grooming. Thanks for the correction!

**Comment by GuillaumeGomez** [comment]
I'm the one who wrote this lint and it seems to be working correctly to me as well...

**Comment by abonander** [comment]
Except that they're reexported from a module marked `#[allow(missing_docs)]` which I would expect *not* to require documentation on items reachable inside it.

Moreover, `#[allow(missing_docs)]` on the reexport does nothing but is not at least marked as an unused attribute.
